File: CSharp/Core/NLog.Target.LogView.Core/Extensions/ServicesExtension.cs
using LogViewer.Core;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;
using NLog.Extensions.Logging;
using MsLogLevel = Microsoft.Extensions.Logging.LogLevel;

namespace NLog.Target.LogView.Core.Extensions;

public static class ServicesExtension
{
    public static ILoggingBuilder AddNLogTargets(this ILoggingBuilder builder, IConfiguration config)
    {
        LogManager
            .Setup()
            // Register custom Target
            .SetupExtensions(extensionBuilder =>
                extensionBuilder.RegisterTarget<DataStoreLoggerTarget>("DataStoreLogger"));

        builder
            .ClearProviders()
            .SetMinimumLevel(MsLogLevel.Trace)
            // Load NLog settings from appsettings*.json
            .AddNLog(config,
                // custom options for capturing the EventId information
                new NLogProviderOptions
                {
                    // https://nlog-project.org/2021/08/25/nlog-5-0-preview1-ready.html#nlogextensionslogging-changes-capture-of-eventid
                    IgnoreEmptyEventId = false,
                    CaptureEventId = EventIdCaptureType.Legacy
                });

        return builder;
    }

    public static ILoggingBuilder AddNLogTargets(this ILoggingBuilder builder, IConfiguration config, Action<DataStoreLoggerConfiguration> configure)
    {
        builder.AddNLogTargets(config);
        builder.Services.Configure(configure);
        return builder;
    }
}

File: VB/Core/NLog.Target.LogViewVB.Core/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices
Imports LogViewerVB.Core
Imports Microsoft.Extensions.Configuration
Imports Microsoft.Extensions.DependencyInjection
Imports Microsoft.Extensions.Logging
Imports NLog.Extensions.Logging
Imports MsLogLevel = Microsoft.Extensions.Logging.LogLevel

Public Module ServicesExtension

    <Extension>
    Public Function AddNLogTargets(builder As ILoggingBuilder, config As IConfiguration) As ILoggingBuilder

        LogManager _
            .Setup() _
            .SetupExtensions(
                Sub(extensionBuilder) _
                    extensionBuilder.RegisterTarget(Of DataStoreLoggerTarget)("DataStoreLogger"))

        ' Load NLog settings from appsettings*.json
        ' custom options for capturing the EventId information
        ' https://nlog-project.org/2021/08/25/nlog-5-0-preview1-ready.html#nlogextensionslogging-changes-capture-of-eventid
        builder _
            .ClearProviders() _
            .SetMinimumLevel(MsLogLevel.Trace) _
            .AddNLog(config,
                New NLogProviderOptions With
                {
                    .IgnoreEmptyEventId = False,
                    .CaptureEventId = EventIdCaptureType.Legacy
                })

        Return builder

    End Function

    <Extension>
    Public Function AddNLogTargets(builder As ILoggingBuilder, config As IConfiguration, configure As Action(Of DataStoreLoggerConfiguration)) As ILoggingBuilder

        builder.AddNLogTargets(config)
        builder.Services.Configure(configure)
        Return builder

    End Function

End Module